operator/v1/types_csi_cluster_driver.go

```diff
@@ -96,6 +96,55 @@ type ClusterCSIDriverSpec struct {
 	// The current default behaviour is Managed.
 	// +optional
 	StorageClassState StorageClassStateName `json:"storageClassState,omitempty"`
+
+	// driverConfig can be used to specify platform specific driver configuration.
+	// When omitted, this means no opinion and the platform is left to choose reasonable
+	// defaults. These defaults are subject to change over time.
+	// +optional
+	DriverConfig CSIDriverConfigSpec `json:"driverConfig,omitempty"`
+}
+
+// CSIDriverType indicates type of CSI driver being configured.
+// +kubebuilder:validation:Enum=vsphere
+type CSIDriverType string
+
+const (
+	VSphereDriverType CSIDriverType = "vsphere"
+)
+
+// CSIDriverConfigSpec defines configuration spec that can be
+// used to optionally configure a specific CSI Driver.
+// +union
+type CSIDriverConfigSpec struct {
+	// driverType indicates type of CSI driver for which the
+	// driverConfig is being applied to.
+	//
+	// Valid values are:
+	//
+	// * vsphere
+	//
+	// Allows configuration of vsphere CSI driver topology.
+	//
+	// +kubebuilder:validation:Required
+	// +unionDiscriminator
+	DriverType CSIDriverType `json:"driverType"`
+
+	// vsphere is used to configure the vsphere CSI driver.
+	// +optional
+	VSphere *VSphereCSIDriverConfigSpec `json:"vsphere,omitempty"`
```

Comment:
If the right `CSIDriverType` is  `"VSphere"`, then this must be `json:"vSphere`

Reply:
fixed.